File: day_1_2022/first_star/zig/src/main.zig
const std = @import("std");

pub fn main() !void {
    var arena = std.heap.ArenaAllocator.init(std.heap.page_allocator);
    defer arena.deinit();

    const file_path = "../../input.txt";
    const file = try std.fs.cwd().openFile(file_path, .{});
    defer file.close();
    var reader = file.reader();

    var maxCalories: u64 = 0;
    var currentCalories: u64 = 0;
    var buffer: [4096]u8 = undefined;
    while (try reader.readUntilDelimiterOrEof(&buffer, '\n')) |line| {
        if (line.len == 0) {
            maxCalories = @max(maxCalories, currentCalories);
            currentCalories = 0;
        } else {
            currentCalories += std.fmt.parseInt(u64, line, 10) catch continue;
        }
    }
    maxCalories = @max(maxCalories, currentCalories);
    const stdout = std.io.getStdOut().writer();
    try stdout.print("{}\n", .{maxCalories});
}

File: day_1_2022/first_star/rust/src/main.rs
use std::{fs::File, io::{BufRead, BufReader}, str::FromStr};

fn main() {
    let input = "../../input.txt";
    let file = File::open(input).expect("Unable to open file");
    let reader = BufReader::new(file);

    let mut max_calories = 0;
    let mut current_calories = 0;

    for line in reader.lines() {
        let line = line.expect("Unable to read line");
        if line.is_empty() {
            max_calories = max_calories.max(current_calories);
            current_calories = 0;
        } else {
            current_calories += i32::from_str(&line).expect("Unable to parse number");
        }
    }

    max_calories = max_calories.max(current_calories);

    println!("{}", max_calories);
}